cobigen/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/GenerationIT.java

```diff
@@ -65,16 +71,86 @@ public void testOverrideMergeStrategy() throws Exception {
     File target = new File(folder, "generated.txt");
     FileUtils.write(target, "base");
 
-    CobiGen cobigen = CobiGenFactory.create(new File(testFileRootPath + "overrideMergeStrategy").toURI());
+    CobiGen cobigen = CobiGenFactory.create(new File(testFileRootPath + "overrideMergeStrategy").toURI(), true);
     List<TemplateTo> templates = cobigen.getMatchingTemplates(input);
-    assertThat(templates).hasSize(1);
+    assertThat(templates).hasSize(2);
 
     GenerationReportTo report = cobigen.generate(input, templates.get(0), Paths.get(folder.toURI()));
 
     assertThat(report).isSuccessful();
     assertThat(target).hasContent("overwritten");
   }
 
+  /**
+   * Tests that multiple template sets with templates, increments and triggers get read and generate of all matching
+   * (java) templates is successful
+   *
+   * @throws Exception test fails.
+   */
+  @Test
+  public void testReadMultipleTemplateSetsWithJavaInput() throws Exception {
+
+    CobiGen cobigen = CobiGenFactory.create(new File(testFileRootPathTemplateSetXml + "template-sets").toURI());
+
+    Object input = cobigen.read(
+        new File("src/test/java/com/devonfw/cobigen/systemtest/testobjects/io/generator/logic/api/to/InputEto.java")
+            .toPath(),
+        Charset.forName("UTF-8"), getClass().getClassLoader());
+
+    File folder = this.tmpFolder.newFolder("GenerationTest");
+    File target = new File(folder, "generated.txt");
+    FileUtils.write(target, "base");
+
+    List<TemplateTo> templates = cobigen.getMatchingTemplates(input);
+    List<IncrementTo> increments = cobigen.getMatchingIncrements(input);
+    List<String> triggersIds = cobigen.getMatchingTriggerIds(input);
+    assertThat(templates).hasSize(4);
+    assertThat(increments).hasSize(4);
+    assertThat(triggersIds).hasSize(4);
+
+    GenerationReportTo report1 = cobigen.generate(input, templates.get(0), Paths.get(folder.toURI()));
+    GenerationReportTo report2 = cobigen.generate(input, templates.get(1), Paths.get(folder.toURI()));
+    GenerationReportTo report3 = cobigen.generate(input, templates.get(2), Paths.get(folder.toURI()));
+    GenerationReportTo report4 = cobigen.generate(input, templates.get(3), Paths.get(folder.toURI()));
+
+    assertThat(report1).isSuccessful();
+    assertThat(report2).isSuccessful();
+    assertThat(report3).isSuccessful();
+    assertThat(report4).isSuccessful();
+  }
+
+  /**
+   * Tests that multiple template sets with templates, increments and triggers get read and generate of one matching
```

Comment:
Does this Test really tests mulitple templates but the assert just checks for 1

Reply:
No, this test just checks if a create and generate from a template-sets folder with multiple template sets will only be done for 1 matching template set (OpenAPI)